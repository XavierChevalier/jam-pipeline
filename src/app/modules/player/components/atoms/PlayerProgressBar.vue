<script setup lang="ts">
  import { computed } from 'vue'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'
  import { number } from 'yup'

  const props = defineProps({
    progress: {
      type: Number,
      required: true,
      validator: isPropertyValid(number().integer().min(0)),
    },
  })

  const progressFill = computed(() => `${props.progress}%`)
</script>

<template>
  <div class="overflow-hidden h-1 rounded-full bg-primary/20">
    <div class="h-full bg-primary" :style="{ width: progressFill }" />
  </div>
</template>
