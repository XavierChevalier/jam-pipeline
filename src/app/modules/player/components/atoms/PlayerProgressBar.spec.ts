import * as Stories from './PlayerProgressBar.stories'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'

describe('PlayerProgressBar', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .property<number>('progress', (property) => {
      property
        .itShouldBeDefined()
        .itShouldHaveAValidatorFunction()
        .itShouldSuccessWith(0)
        .itShouldSuccessWith(1)
        .itShouldFailWith('negative number', -1)
    })
})
