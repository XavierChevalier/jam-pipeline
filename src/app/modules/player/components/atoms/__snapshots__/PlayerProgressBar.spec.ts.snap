// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerProgressBar stories snapshots [snapshot] should render the "Progress0" component story correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 0%;"
  />
</div>
`;

exports[`PlayerProgressBar stories snapshots [snapshot] should render the "Progress42" component story correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 42%;"
  />
</div>
`;

exports[`PlayerProgressBar stories snapshots [snapshot] should render the "Progress50" component story correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 50%;"
  />
</div>
`;

exports[`PlayerProgressBar stories snapshots [snapshot] should render the "Progress100" component story correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 100%;"
  />
</div>
`;
